projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -245,86 +229,81 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
+      return false;
     }
-    this.start = 0;
-    this.end = Math.min(this.previewConfig.number as number, this.images.length);
+    return (this.previewConfig?.arrows && this.images && this.end < this.images.length) || !!this.slideConfig?.infinite;
   }
 
   /**
-   * Private method to init both `start` and `end` to the end.
+   * Private method to init previews based on the currentImage and the full array of images.
+   * The current image in mandatory to show always the current preview (as highlighted).
+   * @param InternalLibImage currentImage to decide how to show previews, because I always want to see the current image as highlighted
+   * @param InternalLibImage[] images is the array of all images.
    */
-  private setEndIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = this.images.length - 1 - ((this.previewConfig.number as number) - 1);
-    this.end = this.images.length;
+  private initPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
+    this.setIndexesPreviews(currentImage, images);
+    this.previews = images.filter((img: InternalLibImage, i: number) => i >= this.start && i < this.end);
   }
 
   /**
    * Private method to update both `start` and `end` based on the currentImage.
    */
-  private setIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images || !this.currentImage) {
+  private setIndexesPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
+    if (!this.previewConfig || !images || !currentImage) {
       throw new Error('Internal library error - previewConfig, currentImage and images must be defined');
     }
-    this.start = getIndex(this.currentImage, this.images) - Math.floor((this.previewConfig.number as number) / 2);
-    this.end = getIndex(this.currentImage, this.images) + Math.floor((this.previewConfig.number as number) / 2) + 1;
+    const previewsNumber = this.previewConfig.number as number;
+    let start = getIndex(currentImage, images) - Math.floor(previewsNumber / 2);
+    // start is, at a minimum, the first index
+    if(start < 0) start = 0;
+    // end index
+    let end = start + previewsNumber;
+    // end is, at a maximum, the last index
+    if(end > images.length) {
+      start -= end - images.length;
+      if(start < 0) start = 0; // start is, at a minimum, the first index
+      end = images.length;
+    }
+    this.start = start;
+    this.end = end;
   }
 
   /**
    * Private method to update the visible previews navigating to the right (next).
    */
   private next(): void {
-    if (!this.images) {
+    if (!this.images || !this.previewConfig) {
       throw new Error('Internal library error - images must be defined');
     }
-    // check if nextImage should be blocked
-    if (this.isPreventSliding(this.images.length - 1)) {
```

Comment:
preventSliding depends whether next() or previous() is called, so I moved the isPreventSliding logic within these two methods for better contextualization.